src/js/techreport/combobox.js

```diff
@@ -0,0 +1,249 @@
+          this.selectElement(e.target);
+        }
+      }
+    });
+    input.addEventListener('click', this.showOptions.bind(this));
+    input.addEventListener('input', this.filterOptions.bind(this));
+    input.addEventListener('blur', this.inputBlur.bind(this));
+    input.addEventListener('keydown', this.navigateOptions.bind(this));
+  }
+
+  inputBlur() {
+    this.hideOptions();
+  }
+
+  showOptions() {
+    const input = this.element.querySelector('input[type="text"]');
+    input.setAttribute('aria-expanded', 'true');
+    const listbox = this.element.querySelector('[role="listbox"]');
+    listbox.classList.remove('hidden');
+  }
+
+  hideOptions() {
+    const input = this.element.querySelector('input[type="text"]');
+    input.setAttribute('aria-expanded', 'false');
+    const options = this.element.querySelector('[role="listbox"]');
+    options.classList.add('hidden');
+  }
+
+  filterOptions(e) {
+    const search = e.target.value;
+    const options = this.element.querySelector('[role="listbox"]');
+
+    this.filteredData = this.data.filter(row => {
+      return row.toLowerCase().includes(search.toLowerCase());
+    });
+
+    options.textContent = '';
+    this.updateContent(this.filteredData);
+
+    if(options.classList.contains('hidden')) {
+      this.showOptions();
+    }
+
+    if(this.filteredData.includes(this.focusedOptionStr)) {
+      this.focusedOption = this.filteredData.indexOf(this.focusedOptionStr);
+      this.updateHighlightedOption();
+    } else {
+      this.focusedOption = -1;
+    }
+  }
+
+  navigateOptions(e) {
+    const listbox = this.element.querySelector('[role="listbox"]');
+    const options = this.element.querySelectorAll('[role="option"]');
+
+    if(listbox.classList.contains('hidden')) {
+      this.showOptions();
+    }
+
+    const key = e.keyCode;
+    switch(key) {
+      case this.keyCodes.keyDown:
+        const nextKey = this.focusedOption + 1;
+        if(nextKey < options.length) {
+          this.focusedOption = nextKey;
+          this.updateHighlightedOption();
+        }
+        break;
+      case this.keyCodes.keyUp:
+        const previousKey = this.focusedOption - 1;
+        if(previousKey > -1) {
+          this.focusedOption = previousKey;
+          this.updateHighlightedOption();
+        }
+        break;
+      case this.keyCodes.ESC:
+        this.hideOptions();
+        break;
+      case this.keyCodes.enter:
+        e.preventDefault();
+        const listbox = this.element.querySelector('[role="listbox"]');
+        if(!listbox.classList.contains('hidden')) {
+          const option = options[this.focusedOption];
+          const selected = option.getAttribute('aria-selected');
+          if(selected === 'true') {
+            this.unselectElement(option.dataset.name);
+          } else {
+            this.selectElement(option);
+          }
+        }
+    }
+  }
```

Comment:
Is there a reason we can't use this?

```suggestion
    const key = e.key;
    switch(key) {
      case 'ArrowDown':
        const nextKey = this.focusedOption + 1;
        if(nextKey < options.length) {
          this.focusedOption = nextKey;
          this.updateHighlightedOption();
        }
        break;
      case 'ArrowUp':
        const previousKey = this.focusedOption - 1;
        if(previousKey > -1) {
          this.focusedOption = previousKey;
          this.updateHighlightedOption();
        }
        break;
      case 'Escape':
        this.hideOptions();
        break;
      case 'Enter':
        e.preventDefault();
        const listbox = this.element.querySelector('[role="listbox"]');
        if(!listbox.classList.contains('hidden')) {
          const option = options[this.focusedOption];
          const selected = option.getAttribute('aria-selected');
          if(selected === 'true') {
            this.unselectElement(option.dataset.name);
          } else {
            this.selectElement(option);
          }
        }
    }
  }
```